crates/context_aware_config/src/api/default_config/handlers.rs

```diff
@@ -201,6 +172,118 @@ async fn create(
     Ok(http_resp.json(default_config))
 }
 
+#[put("/{key}")]
+async fn update_config(
+    state: web::Data<AppState>,
+    key: web::Path<DefaultConfigKey>,
+    custom_headers: CustomHeaders,
+    request: web::Json<UpdateReq>,
+    db_conn: DbConnection,
+    #[cfg(feature = "high-performance-mode")] tenant: Tenant,
+    user: User,
+) -> superposition::Result<HttpResponse> {
+    let DbConnection(mut conn) = db_conn;
+    let req = request.into_inner();
+    let key_str = key.into_inner().into();
+    let tags = parse_config_tags(custom_headers.config_tags)?;
+
+    let existing = fetch_default_key(&key_str, &mut conn).map_err(|e| match e {
+        superposition::AppError::DbError(diesel::NotFound) => {
+            bad_argument!(
+                "No record found for {}. Use create endpoint instead.",
+                key_str
+            )
+        }
+        _ => {
+            log::error!("Failed to fetch {key_str}: {e}");
+            unexpected_error!("Something went wrong.")
+        }
+    })?;
+
+    let value = req.value.unwrap_or_else(|| existing.value.clone());
+    let schema = match req.schema {
+        Some(sch) => Value::Object(sch),
+        None => existing.schema.clone(),
+    };
+    let func_name = match req.function_name {
+        Some(Value::String(s)) => Some(s),
+        Some(Value::Null) => None,
+        None => existing.function_name.clone(),
+        Some(_) => return Err(bad_argument!("function_name must be string or null.")),
+    };
```

Comment:
can we please use the same format/type as being used in `dimension/update` endpoint